lib/game.rb

```diff
@@ -0,0 +1,144 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2], [3, 4, 5], [6, 7, 8],
+    [0, 3, 6], [2, 5, 8], [1, 4, 7],
+    [0, 4, 8], [2, 4, 6]
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    # @board.cells.collect{|space| space == "X"}.size.odd? || @board.cells == @board.reset! ? @player_1 : @player_2
+    @board.turn_count.even? ? @player_1 : @player_2
+  end
+
+  def over?
+    @board.full? || self.draw? || self.won?
+  end
+
+  # def won?
+  #   # Why is X the winner?
+  #   WIN_COMBINATIONS.any? do |combo|
+  #     if @board.taken?(combo[0]) && @board.cells[combo[0]] == @board.cells[combo[1]] && @board.cells[combo[1]] == @board.cells[combo[2]]
+  #       return combo
+  #     end
+  #   end
+  # end
+  def won?
+
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```